### DISCUSSION TITLE
useDisclosure in V3

### DISCUSSION THREAD

**Discussion by veloware** [chakra-ui/chakra-ui#9142] [General] [answered]
I was migrating from V2 to V3 and noticed I didn't get any TS issues with my instances of `useDisclosure` for controlling modals/dialogs.  It seems that the `isOpen` can be changed to `open` and can be passed to the `DialogRoot`, and the dialog will open, however the DialogRoot doesn't have an `onClose` prop (but rather `onOpenChange`?)

<img width="845" alt="image" src="https://github.com/user-attachments/assets/10e9391f-8fdc-4bd9-aee5-efbdec8ea85a">

Therefore it looks like `useDisclosure` only partially works with Dialogs at the moment.  But I also notice it has been removed from the docs.  My question is, should I continue to use `useDisclosure` and the interface will be updated to match DialogRootProps (i.e. `onClose` -> `onOpenChange` ), or should I just use a simple boolean state to control Dialogs?

**Comment by segunadebayo** [comment, marked as answer]
`useDisclosure` still works. Pass `open` to `Dialog.Root` and wire `onOpenChange` to `onOpen` / `onClose`.
